gossip/minimal/goshawk.go

```diff
@@ -453,22 +454,14 @@ func (hawk *Goshawk) Fly(ctx context.Context) {
 	}
 
 	// Flush scan state occasionally in case of abrupt termination.
-	ticker := time.NewTicker(hawk.fetchOpts.FlushInterval)
-	go func() {
-		for {
-			select {
-			case <-ctx.Done():
-				return
-			case <-ticker.C:
-				if err := hawk.fetchOpts.State.Flush(ctx); err != nil {
-					// Keep going even if we failed to save current state; it
-					// just means that the next run will repeat checks that have
-					// already been done.
-					glog.Errorf("State flush failed: %v", err)
-				}
-			}
+	go schedule.Every(ctx, hawk.fetchOpts.FlushInterval, func(ctx context.Context) {
```

Comment:
Will this (slightly) change behaviour so that it's now [Flush, wait, Flush, wait...] rather than [wait, Flush, wait, Flush, ...] ?

Reply:
Yes it will. Would you like a sleep introduced, or is it fine to flush straight away?

Reply:
Should be fine as-is.